pkg/reader/usdc_reader.go

```diff
@@ -113,12 +111,32 @@ func NewUSDCMessageReader(
 				lggr:           lggr,
 				contractReader: contractReaders[chainSelector],
 				cctpDestDomain: domains,
-				boundContract:  contract, // TODO: this is not needed if we switch to the Extended contract reader.
+			}
+		case sel.FamilySolana:
+			// Bind the TokenPool contract, the contract re-emits the USDC MessageSent event along with other metadata.
+			bytesAddress, err := addrCodec.AddressStringToBytes(token.SourceMessageTransmitterAddr, chainSelector)
```

Comment:
Not sure if it's set for both families, but we will need to have time retention set for these events just to make sure we don't store them forever 

https://github.com/smartcontractkit/chainlink/blob/e6d7c15c29751ccd83e3ffaddce420471fd9bbe4/core/capabilities/ccip/configs/evm/contract_reader.go#L406